File: 797-rabbits-in-forest/rabbits-in-forest.java
import java.util.HashMap;


class Solution {

    public int numRabbits(int[] answers) {

        int res = 0;

        HashMap<Integer,Integer> map = new HashMap();

        for(int num : answers){
            map.put(num, map.getOrDefault(num,0)+1);
        }
        
        for(int key : map.keySet()){
            
            int rabbits = (int) Math.ceil((double) map.get(key)/(key+1));
            res += (key+1)*(rabbits);
        }

        return res;
        

    }
}

File: 797-rabbits-in-forest/rabbits-in-forest.py
class Solution:
    import math
    def numRabbits(self, answers: List[int]) -> int:

        s = {}
        for i in answers:
            if i not in s:
                s[i] = 1
            else:
                s[i] += 1
        res = 0
        for i in s:
            res += (i+1)*(math.ceil(s[i]/(i+1)))
        return res
               